### ISSUE TITLE
Consider using dispatch tables for memory reading and writing

### ISSUE THREAD

**Issue by ghost** [NTmatter/LLBoy#3]
Essentially you can make a table that holds pointers to handling sub-routines for addresses being requested. The GameBoy/GameBoy Color memory map is only 16-bit (albeit with ROM and RAM banking), so it can easily be mapped. This is faster than if/else'ing like a mad-man for address decoding.

**Comment by NTmatter** [comment]
Looks like a good idea. I'll give it a try further down the line.

**Comment by ghost** [comment]
Also the games can execute instructions from the their RAM, and often do for waiting out OAM DMA, so make sure you can successfully cover invalidation of the code cache.

**Comment by NTmatter** [comment]
That's a definite chunk of complexity right there. Odds are that I'll just avoid caching any executable instructions in RAM. The real focus of this project is fiddling around with LLVM to produce a mostly native binary of the ROM. It might be worth using the JIT to handle "live" code, though :)

**Comment by ghost** [comment]
Yeah, still might be worth it to JIT the busy-wait loops that are often found in RAM in many games.

**Comment by NTmatter** [comment]
It'll be interesting to do some profiling on the effects of JIT with cache invalidation compared to interpreted execution.

**Comment by ghost** [comment]
The cases of busy-wait are busy enough to hard-code optimizations upon.
